jmp/_src/loss_scale.py

```diff
@@ -115,10 +115,12 @@ class DynamicLossScale:
   ...   # conditionally update params using grads
   """
   loss_scale: jnp.ndarray
-  counter: jnp.ndarray = np.zeros([], np.int32)
+  counter: jnp.ndarray = dataclasses.field(
```

Comment:
Array scalars are immutable in NumPy, perhaps we can use that instead: `counter: jnp.ndarray = np.int32(0)`. Slightly less verbose than using the factory.

Reply:
The scalar isn't an `ndarray` though: `isinstance(np.int32(0), jnp.ndarray)` is False.

(Incidentally, you may at some point want to modernize your `jnp.ndarray` annotation to the new `jax.Array` annotation.)

Reply:
IIRC `np.ndarray` instances are not instances of `jax.Array`.

We're avoiding using `jnp.zeros` for the default value since this has the side effect of initialising JAX backends at import time.

Lets go with your initial suggestion of a factory function.

Reply:
Okay, I had just changed it to `jnp.zeros` before you sent this comment.  Should I change it back?  It should only run when the factory is run, which is when an actual `DynamicLossScale` object is created.

Reply:
I'll flip it to `np` before merging, just incase someone is constructing `DynamicLossScale` at the module level of their experiment.

Reply:
Okay, sorry about that.  And thanks for the quick turnaround!